src/util.py

```diff
@@ -0,0 +1,78 @@
+import discord
+from collections.abc import Coroutine
+from settings import get_settings
+import enums
+
+settings = get_settings()
+
+
+def log_info_and_embed(embed: discord.Embed, logger, message: str):
+    """
+    Write a log message to both the default logger and add the string to
+    the discord message that will be sent to the logging channel upon command
+    finishing
+    """
+    if embed.description is None:
+        embed.description = ""
+    embed.description += "\n" + message
+    logger.info(message)
+
+
+async def send_dm(member: discord.Member, messageContent: str):
+    channel = await member.create_dm()
+    await channel.send(content=messageContent)
+
+
+# NB this feels very messy, fragile, and easy to mess up
+# There might be a better way to wrap all this in something cleaner
```

Comment:
i think this is the best we'll get with this

Reply:
I updated this to use the context manager pattern, it's now way more straightforward